parsers/IN_WE.py

```diff
@@ -18,153 +18,144 @@
 from parsers.lib.config import refetch_frequency
 from parsers.lib.exceptions import ParserException
 
-IN_WE_PROXY = "https://in-proxy-jfnx5klx2a-el.a.run.app"
-EXCHANGE_URL = f"{IN_WE_PROXY}/InterRegionalLinks_Data.aspx/Get_InterRegionalLinks_Region_Wise?host=https://www.wrldc.in"
-CONSUMPTION_URL = f"{IN_WE_PROXY}/OnlinestateTest1.aspx/GetRealTimeData_state_Wise?host=https://www.wrldc.in"
+"""
+This module is used to read the exchange values on a hourly basis for a day from India West to North, East, South. 
+This module also calculates the total electricity consumption by India West.
+The data is fetched from https://www.wrldc.in using APIs which provide data in JSON format
+India West constitutes of Gujrat, Madhya Pradesh, Chattishgarh, Maharashtra, Goa, DD, DNH
 
-EXCHANGES_MAPPING = {
-    "WR-SR": "IN-SO->IN-WE",
-    "WR-ER": "IN-EA->IN-WE",
-    "WR-NR": "IN-NO->IN-WE",
-}
+These APIs return the data in 24 hour format from midnight 12:00am to current time at which the parser is run  
+"""
 
-KIND_MAPPING = {
-    "exchange": {"url": EXCHANGE_URL, "datetime_column": "lastUpdate"},
-    "consumption": {"url": CONSUMPTION_URL, "datetime_column": "current_datetime"},
-}
 
+IN_WE_PROXY = "https://in-proxy-jfnx5klx2a-el.a.run.app"
+IN_TZ = "Asia/Kolkata"
+
+""" 
+Exchange data parser
+source: https://www.wrldc.in/content/166_1_FlowsonInterRegionalLinks.aspx
+API : https://www.wrldc.in/InterRegionalLinks_Data.aspx/Get_InterRegionalLinks_Data
+"""
+EXCHANGE_URL = f"{IN_WE_PROXY}/InterRegionalLinks_Data.aspx/Get_InterRegionalLinks_Data?host=https://www.wrldc.in"
+
+"""
+Consumption data parser
+source: https://www.wrldc.in/content/164_1_StateScheduleVsActual.aspx
+API: https://www.wrldc.in/OnlinestateTest1.aspx/GetRealTimeData
+"""
+CONSUMPTION_URL = (
+    f"{IN_WE_PROXY}/OnlinestateTest1.aspx/GetRealTimeData?host=https://www.wrldc.in"
+)
 
-def get_date_range(dt: datetime):
-    return pd.date_range(
-        arrow.get(dt).floor("day").datetime,
-        arrow.get(dt).ceil("day").floor("hour").datetime,
-        freq="H",
-    ).to_pydatetime()
+EXCHANGES_MAPPING = {
+    "IN-SO->IN-WE": "WR-SR",
+    "IN-EA->IN-WE": "WR-ER",
+    "IN-NO->IN-WE": "WR-NR",
+}
 
 
-def fetch_data(
-    kind: str | None = None,
-    session: Session | None = None,
-    target_datetime: datetime | None = None,
-) -> dict:
-    """- get production data from wrldc.in
-    - filter all rows with same hour as target_datetime"""
+def fetch_data(url: str, session: Session, target_datetime: datetime) -> dict:
+    """
+    get data from wrldc.in
+    """
     assert target_datetime is not None
-    assert kind is not None
 
     r = session or Session()
-    payload = {"date": target_datetime.strftime("%Y-%m-%d")}
+    payload = {"date": target_datetime.strftime("%Y-%m-%d"), "Flag": 24}
 
-    resp: Response = r.post(url=KIND_MAPPING[kind]["url"], json=payload)
```

Comment:
In the new code I am simply passing the URL as a pram instead of a type and then reading it from he KIND_MAPPING